home-assistant-addon/tests/e2e/jwt-authentication.spec.ts

```diff
@@ -0,0 +1,219 @@
+import { test, expect } from '@playwright/test';
+import { spawn } from 'child_process';
+import { promisify } from 'util';
+import { generateTestToken, generateExpiredToken } from './helpers/jwt-helper';
+
+const sleep = promisify(setTimeout);
+
+/**
+ * Start a container with JWT authentication enabled
+ */
+async function startContainerWithJWT(jwtSecret: string): Promise<{ containerId: string; cleanup: () => Promise<void> }> {
```

Comment:
This function could reuse the existing startContainer function if we add the possibility to supply environment variables to the startContainer function.